### PR TITLE
gameEvents: created `CHALLENGE_UNO` event.

### PR DESCRIPTION
**Pull request by sethdivyansh**
fixes: #145

## Description
- Created `challengeUno.ts`.
- Checked if a player can be challenged.
- Player can be challenged only if `vulnerableToUNO` is equal to that player.
- If the challenge is successful, the player must draw two cards from the deck as a penalty.
- Created tests for the challenge_uno event and its corner cases in `events.test.ts`.

## How to Test
I have created some tests in `events.test.ts`

## Checklist
- [x] I have tested these changes locally.
- [x] I have reviewed the code and ensured it follows the project's coding guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have assigned reviewers to this pull request.

### CODE REVIEW

**Comment by sethdivyansh** [comment]
@kuv2707 events.test.ts is getting failed in CI test, even though it passes successfully in the local.

**Comment by kuv2707** [comment]
Its a typescript error. The Player type has been renamed. Make the changes accordingly

**Comment by sethdivyansh** [comment]
@kuv2707 I have made the changes.

File: backend/src/uno-game-engine/events/challengeUno.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            message: 'Cannot challenge this player',
+        };
+    }
+
+    return { type: 'SUCCESS', message: 'Can challenge UNO' };
+}
+
+export function challengeUNO(game: GameEngine, event: GameEvent): EventResult {
+    assert(event.type === 'CHALLENGE_UNO', 'Invalid event type');
+    const player = getPlayer(game, event.playerId);
```
Comment (kuv2707) [inline backend/src/uno-game-engine/events/challengeUno.ts:22]: event.playerId refers to the id of the player who triggered the event.

We should modify the event type to have a field `challengedPlayerId` inside data (which is no longer optional)

**Review by kuv2707 — CHANGES_REQUESTED** [review]
Looks like we are allowing the player to be able to challenge himself😂
Let's keep that as an easter egg.

**Comment by sethdivyansh** [comment]
> Looks like we are allowing the player to be able to challenge himself😂 Let's keep that as an easter egg.

Oh right! I didn't think of it.
It would be fun to make it an easter egg 🥚😂

File: backend/src/uno-game-engine/events/challengeUno.ts (hunk 1 above)
Comment (sethdivyansh) [reply backend/src/uno-game-engine/events/challengeUno.ts]: Done 👍
Please review the changes.

**Review by kuv2707 — APPROVED** [review]
Looks great!
I really appreciate that you write tests for all the new logic
